### ISSUE TITLE
Hive Support - Problem with Constraint tab in HTML output

### ISSUE THREAD

**Issue by vidyadharms** [schemaspy/schemaspy#275]
In constraints tab, I don't see anything even though I could see nice relationship diagrams in RelationShips tab. We are giving relationship through meta.xml files. Does this have impact or non working of Constrains Tab?

When I am using SchemaSpy for Mysql DB, I could see lot of constraints in Constraints tab but for Hive, I don't see anything in Constraints tab even tt has Primary and foreign key constraints in .meta.xml file...What could be the reason?

**Comment by npetzall** [comment]
This is not hive specifik, and one could argue that from meta xml you can only visualize relationships. But not enforce them which a constraint is.

But I'm guessing there is no view in the report that shows the relationship except for the diagram.
Or are they shown as parant/children in colum tables?

**Comment by vidyadharms** [comment]
I agree with you @npetzall it may not be related to Hive specific but it is related to meta.xml.

It does shows parent/children links in column tables. 

Ideally if you have relationship in diagram that means it has Primary and Foreign key constraints which should be shown in constraints tab.

**Comment by npetzall** [comment]
I'm not totally sure, but schemaspy has something called implied relationships and I don't think they show as constraints. They just indicate that things a related and not that a constraint exists.

And an argument would be that adding non-existing constraints to a report would be confusing.
The constraint might exist but it might be implemented in the application layer. But it's not a constraint that the dbms enforces.

**Comment by npetzall** [comment]
But the counter argument would be that the schema meta is to augment the database and since you define pk and fk a constraint might be implied.

It sort of raises the question should the schema meta be extended with an attribute on fk that has constraint=true/false.

I suspect, but need to investigate if pk/fk retrieved from dbms is verified as enforced/checked.

Mysql has the check constraint option, which means you can have pk/fk without validation.
Hive 2.0 has pk/fk but doesn't support validation.

And I assume that most dbms can turn validation off and on.

**Comment by vidyadharms** [comment]
I am in agreement on validation part, but the meta.xml are generated from Hive metastore only.

**Comment by npetzall** [comment]
Is the hive metastore a 1:1 with the hive database?
When I get the time I'm gonna spin up my hive image and take a look in the hive metastore.

Since hive metastore can be mysql or postgress the fk constraint and validation will exist, since it supported.

**Comment by vidyadharms** [comment]
Weare using mysql as Hive metastore.
